### PR TITLE
AI Tutor: UI to add details about 👍 👎 feedback 

### PR DESCRIPTION
**Pull request by Erin007**
[CT-426](https://codedotorg.atlassian.net/browse/CT-426) partial
Continuation of #59357 

As we continue piloting AI Tutor and work toward quality testing the AI's responses, we want to collect as much user feedback as possible. Therefore, we're going to solicit additional details from the user about why they selected 👍 or 👎 . This PR sets up the frontend UI to collect additional details about 👍 👎 feedback for AI Tutor responses. 

https://github.com/code-dot-org/code-dot-org/assets/12300669/8f2a9b3d-13b8-46d1-97c6-8403c85eea1f

The details are then stored in the database in the `AiTutorInteractionsFeedback` table. 
Screenshot 2024-06-27 at 11 07 29 AM

## Follow-up work
If we want this feature beyond the pilot phase and if so what it looks like etc is still being decided. There's still so much churn on AI Tutor at the moment that I'm getting this work to a "good enough" stopping point so I can pivot to quality testing AI responses. 

If we keep this feature, we'll want follow up PRs to at least address: 
- [ ] should these components be extracted and shared in the AI Component Library?
- [ ] this implementation only saves the feedback if the user clicks "submit", not every time they toggle the 👍 👎 icons; is that the implementation we want or are we concerned about losing feedback? 
- [ ] more gracefully handle when saving feedback details fails 
- [ ] design feedback and whether we want to include checkboxes in addition to (or instead of?) freeform text

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      thumbsUp: feedbackData.thumbsUp,
+      thumbsDown: feedbackData.thumbsDown,
+      details: feedbackDetails,
+    };
+
+    try {
+      await saveFeedback(messageId, feedback);
+      setFeedbackDetails('');
+    } catch (error) {
+      console.log('Error saving feedback.');
```
Comment (Erin007) [inline apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx:41]: - [ ] handle this more gracefully?

File: apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+  };
+
+  return (
+    <>
+      <textarea
+        placeholder={'Tell us more.'}
```
Comment (Erin007) [inline apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx]: - [ ] workshop string & i18n

File: apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        value={feedbackDetails}
+      />
+
+      <Button
+        onClick={() =>
+          handleFeedbackSubmission(feedbackData, feedbackDetails, messageId)
+        }
+        color={buttonColors.purple}
+        disabled={!feedbackDetails}
+        text={'Submit'}
```
Comment (Erin007) [inline apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx]: - [x] i18n

File: apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    try {
+      await saveFeedback(messageId, feedback);
+      setFeedbackDetails('');
+    } catch (error) {
+      console.log('Error saving feedback.');
+    }
+  };
+
+  return (
```
Comment (Erin007) [inline apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx]: - [x] adjust styling of this component to be in-line and less intrusive (e.g. probably smaller)

File: apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        onChange={e => setFeedbackDetails(e.target.value)}
+        value={feedbackDetails}
+      />
+
+      <Button
+        onClick={() =>
+          handleFeedbackSubmission(feedbackData, feedbackDetails, messageId)
+        }
+        color={buttonColors.purple}
+        disabled={!feedbackDetails}
```
Comment (Erin007) [inline apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx]: - [ ] confirm if it's ok to force the user to include details

File: apps/src/aiTutor/views/AssistantMessageFeedback.tsx [hunk 6]
```diff
@@ -18,29 +20,6 @@ const AssistantMessageFeedback: React.FC<AssistantMessageProps> = ({
...
-      setFeedbackState(feedbackData);
-      await saveFeedback(messageId, feedbackData);
-    } catch (error) {
-      setFeedbackState({thumbsUp: null, thumbsDown: null});
-    }
-  };
-
   return (
     <div className={style.feedbackIcons}>
       Was this helpful?
```
Comment (Erin007) [inline apps/src/aiTutor/views/AssistantMessageFeedback.tsx]: - [x] i18n

File: apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        text={'Submit'}
+      />
+    </>
+  );
+};
```
Comment (Erin007) [inline apps/src/aiTutor/views/AssistantMessageFeedbackDetails.tsx:69]: - [x] add a "Clear" option to be more consistent with TA

File: apps/src/aiTutor/views/AssistantMessageFeedback.tsx [hunk 8]
```diff
@@ -2,7 +2,9 @@ import React, {useState} from 'react';
```
Comment (Erin007) [inline apps/src/aiTutor/views/AssistantMessageFeedback.tsx:2]: - [ ] review needs and decide if we should share this component + details with other AI projects

**Review by molly-moen — APPROVED** [review]

**Pull request merged**